app/src/main/java/is/xyz/preference/ConfigEditDialogPreference.kt

```diff
@@ -0,0 +1,56 @@
+package `is`.xyz.preference
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import androidx.preference.Preference
+import `is`.xyz.mpv.R
+import `is`.xyz.mpv.databinding.ConfEditorBinding
+import java.io.File
+
+class ConfigEditDialogPreference(
+    context: Context,
+    attrs: AttributeSet? = null
+) : Preference(context, attrs) {
+    private var configFile: File
+    private lateinit var binding: ConfEditorBinding
+    private lateinit var editText: EditText
+
+    init {
+        isPersistent = false
+
+        // determine where the file to be edited is located
+        val styledAttrs = context.obtainStyledAttributes(attrs, R.styleable.ConfigEditDialog)
+        val filename = styledAttrs.getString(R.styleable.ConfigEditDialog_filename)
```

Comment:
this may be good practice but makes few sense :)

Reply:
the way to store custom preferences could be simplified in some way, but until then this would have to do.